src/main/java/com/aetherteam/aetherii/client/renderer/entity/layers/KirridWoolLayer.java

```diff
@@ -13,58 +13,59 @@
+    public KirridWoolLayer(RenderLayerParent<KirridRenderState, EntityModel<KirridRenderState>> renderer) {
         super(renderer);
     }
 
     @Override
-    public void render(PoseStack poseStack, MultiBufferSource bufferSource, int packedLight, Kirrid kirrid, float limbSwing, float limbSwingAmount, float partialTick, float ageInTicks, float netHeadYaw, float headPitch) {
-        if (kirrid.isInvisible()) {
+    public void render(PoseStack poseStack, MultiBufferSource bufferSource, int packedLight, KirridRenderState kirrid, float netHeadYaw, float headPitch) {
+        if (kirrid.isInvisible) {
             Minecraft minecraft = Minecraft.getInstance();
-            boolean flag = minecraft.shouldEntityAppearGlowing(kirrid);
+            boolean flag = kirrid.appearsGlowing;
             if (flag) {
-                this.getParentModel().prepareMobModel(kirrid, limbSwing, limbSwingAmount, partialTick);
-                this.getParentModel().setupAnim(kirrid, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch);
+                this.getParentModel().setupAnim(kirrid);
                 VertexConsumer consumer = bufferSource.getBuffer(RenderType.outline(this.getTexture(kirrid)));
                 this.getParentModel().renderToBuffer(poseStack, consumer, packedLight, LivingEntityRenderer.getOverlayCoords(kirrid, 0.0F), -16777216);
             }
         } else {
             int i = -1;
-            if (kirrid.hasCustomName() && kirrid.getName().getString().equals("jeb_")) {
+            if (kirrid.customName != null && "jeb_".equals(kirrid.customName.getString())) {
                 int j = 25;
-                int k = kirrid.tickCount / j + kirrid.getId();
-                int l = Kirrid.KirridColor.values().length;
-                int i1 = k % l;
-                int j1 = (k + 1) % l;
-                float f = ((float) (kirrid.tickCount % j) + partialTick) / j;
-                int k1 = Kirrid.getDecimalColor(Kirrid.KirridColor.BY_ID.apply(i1));
-                int l1 = Kirrid.getDecimalColor(Kirrid.KirridColor.BY_ID.apply(j1));
-                i = FastColor.ARGB32.lerp(f, k1, l1);
-                coloredCutoutModelCopyLayerRender(this.getParentModel(), this.getParentModel(), this.getTexture(kirrid), poseStack, bufferSource, packedLight, kirrid, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch, partialTick, i);
-            } else if (kirrid.getColor().isPresent()) {
-                i = Kirrid.getDecimalColor(kirrid.getColor().get());
-                coloredCutoutModelCopyLayerRender(this.getParentModel(), this.getParentModel(), this.getTexture(kirrid), poseStack, bufferSource, packedLight, kirrid, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch, partialTick, i);
+                int k = Mth.floor(kirrid.ageInTicks);
+                int l = k / 25 + kirrid.id;
+                int i1 = DyeColor.values().length;
```

Comment:
The previous color handling from the Kirrid class should be used here, since it has a special case for what color a Kirrid is if it hasn't been dyed.